modules/tinymce/src/plugins/lists/main/ts/core/Util.ts

```diff
@@ -1,3 +1,28 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
 import { HTMLElement } from '@ephox/dom-globals';
+import { Arr } from '@ephox/katamari';
+import Editor from 'tinymce/core/api/Editor';
+import Tools from 'tinymce/core/api/util/Tools';
+import * as NodeType from './NodeType';
+
+export const isCustomList = (list: HTMLElement) => /\btox\-/.test(list.className);
+
+export const listState = (editor: Editor, listName: string, activate: (active: boolean) => void) =>
+  () => {
+    const nodeChangeHandler = (e) => {
+      const tableCellIndex = Arr.findIndex(e.parents, NodeType.isTableCellNode);
+      const parents = tableCellIndex.map((idx) => e.parents.slice(0, idx)).getOr(e.parents);
+      const lists = Tools.grep(parents, NodeType.isListNode);
```

Comment:
I realise it's just a refactoring, but that new `findUntil` method could make this "find list node" until "is table cell"?

In the past I've done this sort of thing with `SelectorFind.ancestor("ol")` using "is table cell" as the `isRoot` parameter.

Reply:
Yeah, I didn't look at this code at all and just moved it. I'll see about cleaning it up.

Reply:
It wasn't the nicest since we need to call activate at all times, but it's still a lot cleaner than it was previously.